src/utils/@reduxjs/toolkit.test.tsx

```diff
@@ -0,0 +1,38 @@
+// eslint-disable-next-line no-restricted-imports
+import { createSlice as createSliceOriginal } from '@reduxjs/toolkit'
+import { createSlice } from 'utils/@reduxjs/toolkit'
+
+describe('type-only test', () => {
+  describe('utils/@reduxjs/toolkit is strictly typed', () => {
+    it('correct usage', () => {
+      expect(
+        createSlice({
+          name: 'account',
+          initialState: {},
+          reducers: {},
+        }),
+      ).toBeDefined()
+    })
+
+    it('detect mistakes', () => {
```

Comment:
I don't mind merging this, but what's a real value of these tests ? For example this test is testing  a case which is impossible to implement, because TS will throw an error. How any of these test can fail ?

Reply:
This ensures that typescript will throw an error. E.g. if we change our type definition or bump typescript, our constraints could become less strict without us noticing otherwise

Reply:
make sense now, thanks!